S/Sundials/Sundials@7/build_tarballs.jl

```diff
@@ -0,0 +1,100 @@
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}" \
+    -DEXAMPLES_ENABLE_C=OFF \
+    -DENABLE_KLU=ON \
+    -DKLU_INCLUDE_DIR="${includedir}/suitesparse" \
+    -DKLU_LIBRARY_DIR="${libdir}" \
+    -DKLU_WORKS=ON \
+    -DENABLE_LAPACK=ON \
+    -DBLAS_LIBRARIES="${BLAS_NAME}" \
+    -DLAPACK_LIBRARIES="${LAPACK_NAME}" \
+    -DLAPACK_WORKS=ON \
+    ..
+make -j${nproc}
+make install
+
+# Move libraries to ${libdir} on Windows
+if [[ "${target}" == *-mingw* ]]; then
+    mv ${prefix}/lib/libsundials_*.${dlext} "${libdir}"
+fi
+"""
+
+# We attempt to build for all defined platforms
+platforms = supported_platforms()
+platforms = expand_gfortran_versions(platforms)
+
+products = [
+    LibraryProduct("libsundials_arkode", :libsundials_arkode),
+    LibraryProduct("libsundials_cvode", :libsundials_cvode),
+    LibraryProduct("libsundials_cvodes", :libsundials_cvodes),
+    LibraryProduct("libsundials_ida", :libsundials_ida),
+    LibraryProduct("libsundials_idas", :libsundials_idas),
+    LibraryProduct("libsundials_kinsol", :libsundials_kinsol),
+    LibraryProduct("libsundials_nvecmanyvector", :libsundials_nvecmanyvector),
+    LibraryProduct("libsundials_nvecserial", :libsundials_nvecserial),
+    LibraryProduct("libsundials_sunlinsolband", :libsundials_sunlinsolband),
+    LibraryProduct("libsundials_sunlinsoldense", :libsundials_sunlinsoldense),
+    LibraryProduct("libsundials_sunlinsolklu", :libsundials_sunlinsolklu),
+    LibraryProduct("libsundials_sunlinsollapackband", :libsundials_sunlinsollapackband),
+    LibraryProduct("libsundials_sunlinsollapackdense", :libsundials_sunlinsollapackdense),
+    LibraryProduct("libsundials_sunlinsolpcg", :libsundials_sunlinsolpcg),
+    LibraryProduct("libsundials_sunlinsolspbcgs", :libsundials_sunlinsolspbcgs),
+    LibraryProduct("libsundials_sunlinsolspfgmr", :libsundials_sunlinsolspfgmr),
+    LibraryProduct("libsundials_sunlinsolspgmr", :libsundials_sunlinsolspgmr),
+    LibraryProduct("libsundials_sunlinsolsptfqmr", :libsundials_sunlinsolsptfqmr),
+    LibraryProduct("libsundials_sunmatrixband", :libsundials_sunmatrixband),
+    LibraryProduct("libsundials_sunmatrixdense", :libsundials_sunmatrixdense),
+    LibraryProduct("libsundials_sunmatrixsparse", :libsundials_sunmatrixsparse),
+    LibraryProduct("libsundials_sunnonlinsolfixedpoint", :libsundials_sunnonlinsolfixedpoint),
+    LibraryProduct("libsundials_sunnonlinsolnewton", :libsundials_sunnonlinsolnewton),
+    # Note: libsundials_generic was renamed to libsundials_core in v7
+    LibraryProduct("libsundials_core", :libsundials_core),
+]
+
+dependencies = [
+    Dependency("CompilerSupportLibraries_jll"),
+    Dependency("libblastrampoline_jll"),
+    Dependency("SuiteSparse_jll"),
+]
+
+# Build the tarballs
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"6", julia_compat="1.10")
```

Comment:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"6", julia_compat="1.10")

```
trailing newline is missing